src/utils/generator.js

```diff
@@ -530,6 +530,7 @@ export class Generator {
 
     for (let propertyBinding of modifiedProperties.keys()) {
       let object = propertyBinding.object;
+      realm.lookasideTable.setPropertyLookaside(output, propertyBinding);
```

Comment:
I don't like the idea of having yet another global thing (the `lookasideTable` in this design) that gets mutated as a side effect of some function.

Is the goal here to eventually avoid having to apply effects in the serializer at all? That would be a good thing. But then rather just store this information as a map in the generator. Then again, that's what effects are...

The way CSE is designed, the visitor would then again have to mutate this global `lookasideTable`. That's fragile. I'd rather have the visitor produce a brand new table, restricted to only what's actually reachable; it can mutate it locally to do CSE, but it effectively becomes immutable once the visitor is done.

Reply:
What I read from this comment is you would like the visitor to return a table with no `set-` methods. That seems reasonable and compatible with this approach. 

I'm avoiding storing the information in the generator because it's easier for me to debug when everything is in a single table, but in the last PR once the serializer bit is done, I could move the information into the generators.

Reply:
I could also change this in a later PR so that at the end of the visitor, it incorporates the information about visited bindings/properties and removes any bindings not visited at the very end of the visitor.

Reply:
> What I read from this comment is you would like the visitor to return a table with no set- methods. That seems reasonable and compatible with this approach.

That would be nice.

> I'm avoiding storing the information in the generator because it's easier for me to debug when everything is in a single table

Doesn't really have to be in the generator. But I really don't think that it should be a global table in the realm that is mutated as a side effect calling a function.

> removes any bindings not visited 
I don't want that the visitor mutates any data structures. It could take one table in, and produce another one. But it should mutate anything given to it. (I know that today it does, but I want to move away from that.)